streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -1118,6 +1117,9 @@ boolean sendingOldValueEnabled() {
             ),
             resultStore
         );
+        resultNode.setStoreNeeded(materializedInternal.queryableStoreName() != null);
+        resultProcessorSupplier.setTableProcessorNode(resultNode);
```

Comment:
`KTableSource#materialize()` should be good enough -- it will set the `queryableName` (AFAIK, this won't make the store really queryable, but it's just a proxy to enforce materialization).

Reply:
+1.